### DISCUSSION TITLE
JSDoc - Returns PrismaClient Type

### DISCUSSION THREAD

**Discussion by brunocamboim** [prisma/prisma#17794] [Q&A]
Hi everyone.

I have a schema and I use it inside a function, like that:
`
// dependencies
const { PrismaClient: MongoClient } = require('../../prisma/generated/mongoSchema');

// envs
const { NODE_ENV } = process.env;
const enableLog    = NODE_ENV !== 'production' ? [{ emit: 'stdout', level: 'query' }] : [];

let prismaConnectionMongo;

/**
 *
 * @returns {MongoClient}
 */
module.exports.prismaConnectionMongo = () => {
  if (!prismaConnectionMongo) {
    prismaConnectionMongo = new MongoClient({
      datasources: {
        db: {
          url: process.env.DATABASE_MONGO_URL
        }
      },

      log: enableLog
    });
  }

  return prismaConnectionMongo;
};
`

When I import and use it, it returns as an array:
![image](https://user-images.githubusercontent.com/33419444/217367397-e483ac4a-9d28-4659-8653-417d8a82ad77.png)

What I need is that (but without instantiate):
![image](https://user-images.githubusercontent.com/33419444/217367506-d9479226-2d2e-4060-a2e7-24aab181f282.png)

What are the solutions for that?

Thanks!

**Comment by nurul3101** [comment]
Hello @brunocamboim 👋 

From the code that you have shared, `prismaConnectionMongo` is a function that is returned, so you are getting the apply, arguments etc fields in auto-completion.

Can you elaborate on what return type and auto-completion prompts are you expecting?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
